indico/modules/events/persons/controllers.py

```diff
@@ -63,62 +76,84 @@ def get_persons(self):
         event_strategy.joinedload('person').joinedload('user')
 
         chairpersons = {link.person for link in self.event.person_links}
-        persons = defaultdict(lambda: {'session_blocks': defaultdict(dict),
-                                       'contributions': defaultdict(dict),
-                                       'subcontributions': defaultdict(dict),
-                                       'abstracts': defaultdict(dict),
-                                       'roles': defaultdict(lambda: False)})
+        persons = defaultdict(lambda: {'roles': OrderedDict(), 'registrations': []})
 
-        event_persons_query = (self.event.persons
+        event_persons_query = (db.session.query(EventPerson, Registration)
```

Comment:
Is using a single query for this the best idea? Seems cleaner to split it in two queries. Especially since we do want to show entries where there is only a registration but no `EventPerson`. BTW, we also want to show people who have neither but are in an `EventRole`.

Reply:
Do we want to show all the registrants even if they don't belong to any role?
I've made this query together with @pferreir and in that moment we thought that having results as a tuple of  `(EventPerson, Registration)` was the best idea.

Reply:
OK, if we don't want to show registrants without other roles then this makes sense

Reply:
Yes I agree, I missed those (`EventRole.members`)